server/main.lua

```diff
@@ -28,6 +28,45 @@ CreateThread(function()
     end
 end)
 
+Citizen.CreateThread(function()
+    if Config.CalculateSavingsInterest then
+        local shouldCalculate = false
+        local lastTime = exports.oxmysql:executeSync('SELECT * FROM interest_statements ORDER BY id DESC LIMIT 0, 1')
+        if lastTime[1] then
+            local lastCalculate = lastTime[1].timestamp
+            local diff = os.difftime(os.time(), lastCalculate)
+            local twentyFourHours = 24 * 60 * 60
+            if diff > twentyFourHours then
+                shouldCalculate = true
+            end
+        end
+        
+        if shouldCalculate then
+            local interestAccounts = exports.oxmysql:executeSync('SELECT * FROM bank_accounts', {})
+            if interestAccounts then
+                for i,v in pairs(interestAccounts) do
+                    if v.account_type == 'Savings' then
+                        local balance = v.amount 
+                        local interestAmount = 2 -- in percent
```

Comment:
Should probably just change the interest config to config.interestAmount or something and then allow it to be set to 0. If its 0 you can change the check on line 32.